### ISSUE TITLE
Missing header file with use_frameworks! 

### ISSUE THREAD

**Issue by bensmiley** [CocoaPods/CocoaPods#6526]
# Report

## What did you do?

I created a new podspec file which contains some dependencies. 

```
Pod::Spec.new do |s|
  s.name             = "ChatSDK"
  s.version          = "4.2.5"
  s.summary          = "Chat SDK - Mobile messaging framework for iOS"
  s.homepage         = "http://chatsdk.co"
  s.license          = 'MIT'
  s.author           = { "Ben Smiley" => "ben@chatsdk.co" }
  s.source           = { :git => "https://github.com/chat-sdk/chat-sdk-ios.git", :tag => s.version.to_s }

  s.platform     = :ios, '7.0'
  s.requires_arc = true
  
  s.subspec 'Core' do |core|

    core.source_files = ['Core/**/*']

    core.dependency 'Firebase/Core'
    core.dependency 'Firebase/Database'
    core.dependency 'Firebase/Storage'
    core.dependency 'Firebase/Auth'
  end
end
```

I created a new Xcode project with Xcode 8.2 and added a podfile.

```
source 'https://github.com/CocoaPods/Specs.git'

use_frameworks!

target 'PodTest' do
  
  pod "ChatSDK/Core", :path => "../"
	
end
```

I ran `pod install` using Cocoapods 1.2.0.  

## What did you expect to happen?

I expected the pods to install correctly and for the project to compile. 

## What happened instead?

There is an error message `FirebaseAnalytics/FirebaseAnalytics.h file not found`. This prevents the project from compiling. However, it works if I remove the `use_frameworks!` flag or if I add the Firebase dependencies directly to the Podfile. 

### Stack

```
   CocoaPods : 1.2.0
        Ruby : ruby 2.0.0p648 (2015-12-16 revision 53162) [universal.x86_64-darwin16]
    RubyGems : 2.0.14.1
        Host : Mac OS X 10.12.2 (16C67)
       Xcode : 8.2 (8C38)
         Git : git version 2.10.1
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ dee0949e9974fff9df1e6dcacec3b0f1675165df
               master_old - https://github.com/CocoaPods/Specs.git @ 55393b63a3da4576423ef07b8d1f6f244594638d
               private - https://bensmiley@bitbucket.org/bensmiley/pods.git @ 4f14e0b26b6c6e249d49c8fa37d0eba40dde3d99
```

### Installation Source

```
Executable Path: /usr/local/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.1
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.0.0
cocoapods-trunk       : 1.1.2
cocoapods-try         : 1.1.0
```

### Podfile

```ruby
source 'https://github.com/CocoaPods/Specs.git'

use_frameworks!

target 'PodTest' do
  
  pod "ChatSDK/Core", :path => "../"
		
end
```

## Project that demonstrates the issue

https://www.dropbox.com/s/49xvnx3341k64hi/PodTest.zip?dl=0

**Comment by bensmiley** [comment]
What I've found is that in this situation, the dependent frameworks aren't linked against the ChatSDK framework. This means that any code inside the ChatSDK framework can't access the Firebase libraries. 

There is a workaround which involves adding the frameworks and linker flags manually: 

```
    s.pod_target_xcconfig = { 
        "OTHER_LDFLAGS" => '$(inherited) -framework "FirebaseDatabase" -framework "FirebaseCore" -framework "FirebaseAuth" -framework "FirebaseStorage" -framework "FirebaseInstanceID" -framework "FirebaseAnalytics" -framework "FirebaseDatabase"', 
        "CLANG_ALLOW_NON_MODULAR_INCLUDES_IN_FRAMEWORK_MODULES" => 'YES',
        "FRAMEWORK_SEARCH_PATHS" => '$(inherited) "${PODS_ROOT}/FirebaseAuth/Frameworks" "${PODS_ROOT}/FirebaseCore/Frameworks" "${PODS_ROOT}/FirebaseDatabase/Frameworks" "${PODS_ROOT}/FirebaseInstanceID/Frameworks" "${PODS_ROOT}/FirebaseStorage/Frameworks" "${PODS_ROOT}/FirebaseAnalytics/Frameworks"' 
	}
```

This isn't ideal because it means the frameworks are linked to the project twice which causes warnings in Xcode when the project runs.

**Comment by benasher44** [comment]
Hi @bensmiley! Is PodTest a framework/library-type target?

**Comment by bensmiley** [comment]
No, it's a standard app target - to build the project to run on a device.

**Comment by dnkoutso** [comment]
Definitely confirmed. Need to see what exactly is wrong here.

**Comment by dnkoutso** [comment]
So this is a bit difficult but I think CocoaPods should do the right thing. 

Basically your dynamic framework target "ChatSDK" should be the only one responsible for linking and having the framework search paths for the Firebase vendored static framework libraries. This means the app should not include the framework search paths or the linker flags for the Firebase stuff otherwise you end up with duplicate symbols when you launch the app.

@bensmiley this will be a bit more complicated to solve but I will try to get it in for 1.2.1 as its a great fix around vendored static frameworks and vendored static libraries.

**Comment by bensmiley** [comment]
@dnkoutso That sounds great. In the meantime adding the xcconfig settings does allow the project to compile. The only issue we are seeing is that it creates two separate instances of the Firebase libraries. That means that there can be some problems with singletons because if you use a Firebase singleton from the main project, it will be a different instance than if you used one from the pod's code.

**Comment by bensmiley** [comment]
We've also noticed other issues around the use_frameworks! functionality. We have a modular project and we would like to be able to do the following:

Have multiple podspecs each with subpods. For example:

```
ChatSDK
   - Core
   - UI
        - dependency 'ChatSDK/Core'
        - dependency 'Firebase'
   - Other subspecs

ChatSDKModules
    - Module 1
        - dependency 'ChatSDK/UI'
    - Module 2
```

We have noticed that with the use frameworks flag, this kind of structure creates a lot of issues. Firstly, when importing modules with Swift, we were getting errors saying "Cannot build module..." so we've stopped using subspecs. We've also noticed that dependencies between modules don't work very well so Module 1 isn't able to access the Firebase libraries. 

I'm not an expert in the Xcode build process but in using Cocoapods it feels like the above requirements should just work. I can understand that it may be complicated to implement but it would be useful if there were a document which detailed the limitations of Cocoapods so that we could plan the structure of our project accordingly.

**Comment by crazyjooe** [comment]
Struggling with the same problems as @bensmiley. Having tons of dependancy issues since I switched to use_frameworks! with modular project.

**Comment by suniljaiswal2014** [comment]
Upgrade cocoapods to 1.3.0
sudo gem uninstall cocoapods
sudo gem install cocoapods --pre

**Comment by arkilis** [comment]
@suniljaiswal2014  I am using cocoapods to 1.3.1, issue does exists.

**Comment by dnkoutso** [comment]
Keep
